### DISCUSSION TITLE
Design refactoring

### DISCUSSION THREAD

**Discussion by Benji377** [Benji377/OpenMusic#96] [Ideas]
We already did some serious refactoring referring to discussion #78, but we only did it on the coding level. We basically did it for us and future contributors.
What I think we need to do is a design change. Don't get me wrong, I like the design how it is now, but it's just not good enough. 
First of all, the buttons are messy. Not their appearance, but how they are placed, it just doesn't feel right or comfortable. Then the red color is not really a good choice, it's a screaming "bad" color, red is the equivalent of failure. Another thing is the background, it's just an image I got from the internet randomly. What we need is some sort of background that we can change programmatically.
So here is a list of things we need to change before starting to solve issue #95:
- Rearranging buttons to make it feel more comfortable
- Changing the background from an image to color. I'm thinking of a library that could generate some color mix
- Remove unnecessary UI elements
- Rearranging menu elements

Now I know that I am not the best at explaining things, but fortunately, I can show you some references of music player that has just what I would like to have:

![Screenshot_20210714-215822_Samsung Music](https://user-images.githubusercontent.com/50681275/125685921-c49ecdc7-e8ac-4f2b-8791-7c2a21607b5a.jpg)
As you can see, there is a menu at the top where you can select which fragment you want to look at. This might be super cool if we want to display settings, playlists, and other things on different fragments. I just think it looks cool.
Another thing to notice is the search icon. I definitely want to implement that somewhere in the future.

![Screenshot_20210714-215807_Samsung Music](https://user-images.githubusercontent.com/50681275/125685912-c1d78ef5-1569-482a-a0c0-de3f71315220.jpg)
This is the player fragment basically. As you can see, the background is not a fixed image, but it changes depending on if the song has an image and what time of the day it is. The buttons are also arranged beautifully and clean. This can easily be done by rearranging the constraints in the fragments.

This is the big plan, let me know if you some better suggestions or in general what you think of the idea

**Comment by squivix** [comment]
I don't know much about design myself but I agree with you that we should let the user pick their own colors plus the day/night theme. That's been on my mind for a while so hopefully I'll work on it after a few minor changes I wanna make to the DirBrowser fragment.

As for all the tabs on the main activity, I think that will have to wait until we've made some big changes to the model of the app to get more meta-data from files such as albums, artists, genres, etc and after adding the playlist functionality. Otherwise we won't have anything to put in those tabs :-/

The way we search for files right now is discouraged by the docs. We should be using the MediaStore API. It's a major shift, so I've been delaying it, but we will have to do it sooner or later if we want to get more information out of files and if we want to avoid the ScopedStorage problems of Android 11+ previously discussed [here](https://github.com/Benji377/SocyMusic/issues/80). I'm not familiar with API so I'll have to do some more reading before I'm ready to implement it.

**Comment by squivix** [comment]
I approve of the newest player. It looks great :D

**Reply by Benji377** [reply]
Hahaha perfect. Now my only concern are playlists and the background

**Reply by squivix** [reply]
The app is looking good, Benji! I'm impressed by the work you did on the themes :D

**Reply by Benji377** [reply]
Thank you! I just want to make one more improvement to it. I want to generate the background automatically, which would add much more customization with a lot more colors.

**Reply by Benji377** [reply]
For example, what do you think if we replace the visualizer we have now with [this new one](https://github.com/Cleveroad/WaveInApp) and set it as our background? We can set the color of the waves dynamically trough the theme.

**Reply by squivix** [reply]
I don't know. I like our current visualizer but that may be because I'm used to it. I'll leave the design decisions to you, but I would say that it's not very important now. We've got plenty of very basic features that should be worked out first.

Also we haven't finished theming the app yet! DirBrowserFragment is red no matter what theme you have. I have no idea how to do that so I'll leave it to you :-|

**Reply by Benji377** [reply]
Oh damn, I forgot about that one. Actually, all I did at first was set the theme manually when the app started, and if it changed it would simply reload the app (MainActivity). But then I noticed that some pieces of the app just didn't care about theme changes, for example, the Preferences, which is really strange. I will look into that and see if I need to just manually force the theme change there too, thanks for pointing it out.
